tensorflow/python/ops/array_ops.py

```diff
@@ -470,7 +470,7 @@ def _SliceHelper(tensor, slice_spec, var=None):
     else:
       begin.append(s)
       end.append(s + 1)
-      strides.append(1)
+      strides.append(gen_math_ops.cast(1, s.dtype))
```

Comment:
please use constant_op.constant(1, s.dtype) or np.array(1, dtype=s.dtype.as_numpy_dtype)  (the latter is probably better!)